legend-engine-test-runner-mapping/src/main/java/org/finos/legend/engine/testable/mapping/extension/MappingTestRunner.java

```diff
@@ -125,33 +186,113 @@ public List<TestResult> executeTestSuite(Root_meta_pure_test_TestSuite testSuite
         }
         finally
         {
-            List<Closeable> runtimeCloseables = connections.stream().map(x -> x.getTwo()).flatMap(Collection::stream).collect(Collectors.toList());
-            if (runtimeCloseables != null)
+            this.closeConnections(connections);
+        }
+        return results;
+    }
+
+
+    public List<TestResult> executeMappingQueryTestSuite(MappingQueryTestSuite mappingQueryTestSuite, List<String> atomicTestIds, MappingTestRunnerContext context)
+    {
+        List<org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult> results = Lists.mutable.empty();
+        try
+        {
+            PureModel pureModel = context.modelPairs.getOne();
+            LambdaFunction<?> pureLambda = HelperValueSpecificationBuilder.buildLambda(mappingQueryTestSuite.query, new CompileContext.Builder(pureModel).build());
+            List<MappingDataTest> dataTests = this.validateMappingSuiteTests(mappingQueryTestSuite, MappingDataTest.class);
+            for (MappingDataTest dataTest : dataTests)
             {
-                runtimeCloseables.stream().forEach(closeable ->
+                if (atomicTestIds.contains(dataTest.id))
                 {
-                    try
-                    {
-                        closeable.close();
-                    }
-                    catch (IOException e)
-                    {
-                        LOGGER.warn("Exception occurred closing closeable resource" + e);
-                    }
-                });
+                    org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult testResult = executeMappingDataTest(pureLambda, dataTest, context);
+                    testResult.testable = getElementFullPath(pureMapping, pureModel.getExecutionSupport());
+                    testResult.testSuiteId = mappingQueryTestSuite.id;
+                    results.add(testResult);
+                }
             }
         }
+        catch (Exception e)
+        {
+            throw new RuntimeException("Exception occurred executing service test suites.\n" + e);
```

Comment:
Do we catch these exceptions anywhere to wrap in TestError?

Reply:
We catch the errors within each test run. This would happen i guess in the setup functions when preparing to run the suite. How would we create a testerror from there if a testId is needed. 


I see we do the same thing in serviceTestRunner.